cmd/swarm/upload_test.go

```diff
@@ -101,16 +121,207 @@ func testCLISwarmUp(toEncrypt bool, t *testing.T) {
 		if string(reply) != data {
 			t.Fatalf("expected HTTP body %q, got %q", data, reply)
 		}
+		log.Info("verifying uploaded file using `swarm down`")
+		//try to get the content with `swarm down`
+		tmpDownload, err := ioutil.TempDir("", "swarm-test")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer os.Remove(tmpDownload)
+
+		bzzLocator := "bzz:/" + hash
+		flagss := []string{}
+		flagss = []string{
+			"--verbosity", "5",
+			"--bzzapi", cluster.Nodes[0].URL,
+			"down",
+			bzzLocator,
+			tmpDownload,
+		}
+		if toEncrypt {
+			flagss = []string{
+				"--verbosity", "5",
+				"--bzzapi", cluster.Nodes[0].URL,
+				"down",
+				bzzLocator,
+				tmpDownload}
+		}
+		down := runSwarm(t, flagss...)
+		down.ExpectExit()
+
+		fi, err := os.Stat(path.Join(tmpDownload, hash))
+		if err != nil {
+			t.Fatalf("could not stat path: %v", err)
+		}
+
+		switch mode := fi.Mode(); {
+		case mode.IsRegular():
+			if file, err := swarm.Open(path.Join(tmpDownload, hash)); err != nil {
+				t.Fatalf("encountered an error opening the file returned from the CLI: %v", err)
+			} else {
+
+				ff := make([]byte, len(data))
+				io.ReadFull(file, ff)
+				buf := bytes.NewBufferString(data)
+
+				if !bytes.Equal(ff, buf.Bytes()) {
+					t.Fatalf("retrieved data and posted data not equal!")
+				}
+			}
+		default:
+			t.Fatalf("this shouldnt happen")
+		}
+	}
+
+	timeout := time.Duration(2 * time.Second)
+	httpClient := http.Client{
+		Timeout: timeout,
 	}
 
-	// get an non-existent hash from each node
+	// try to squeeze a timeout by getting an non-existent hash from each node
 	for _, node := range cluster.Nodes {
-		res, err := http.Get(node.URL + "/bzz:/1023e8bae0f70be7d7b5f74343088ba408a218254391490c85ae16278e230340")
+		_, err := httpClient.Get(node.URL + "/bzz:/1023e8bae0f70be7d7b5f74343088ba408a218254391490c85ae16278e230340")
+		// we're speeding up the timeout here since netstore has a 60 seconds timeout on a request
```

Comment:
There should be a request header / query parameter (e.g. `GET /bzz:/<hash>/?timeout=1s`), although this seems unrelated to the addition of `swarm down`?

Reply:
unrelated. this is due to the fact that the netstore timeout is set to 60 seconds. the solution in this case would be to pull the timeout in the netstore to a config flag, then we could instantiate the tests with lower timeouts (and then i'd get rid of this)